### ISSUE TITLE
Delete Orphaned Routes gives 400 for routes with service instance associations

### ISSUE THREAD

**Issue by twoseat** [cloudfoundry/cli#989]
- Command  
  cf delete-orphaned-routes
- What occurred, incl. error and/or stack trace
  I created two routes, one associated with an app, the other with a service:

```
space          host     domain            port   path   type   apps     service
pcfdev-space   my-app   local.pcfdev.io                        my-app
pcfdev-space   pebble   example.com                                     testups
```

I would expect delete orphaned routes to no-op, as no routes are orphaned. Instead it gives an error:

```
FAILED
Server error, status code: 400, error code: 10006, message: Please delete the service_instance associations for your routes.
```
- What you expected to occur
  No deletion and no error
- CLI Version  
  6.22.2+a95e24c-2016-10-27
- CC API Endpoint Version  
  2.58.0
- CF Trace, if issue relates to HTTP requests or strange behavior  
  See https://gist.github.com/twoseat/7dab66b7aa85074feacec76d636cf9c3
- Platform & Shell Details  
  macOS Sierra and stock Terminal
- Any other relevant information
  Running against pcfdev 0.21
  Note that as part of the fix the help text should be changed from:
  `Delete all orphaned routes (i.e. those that are not mapped to an app)`
  to something like:
  `Delete all orphaned routes (i.e. those that are not mapped to an app or service)`

**Comment by twoseat** [comment]
* Command  
cf delete-orphaned-routes

* What occurred, incl. error and/or stack trace
I created two routes, one associated with an app, the other with a service:
```
space          host     domain            port   path   type   apps     service
pcfdev-space   my-app   local.pcfdev.io                        my-app
pcfdev-space   pebble   twoseat.com                                     testups
```

I would expect delete orphaned routes to no-op, as no routes are orphaned. Instead it gives an error:
```
FAILED
Server error, status code: 400, error code: 10006, message: Please delete the service_instance associations for your routes.
```

* What you expected to occur
No deletion and no error

* CLI Version  
6.22.2+a95e24c-2016-10-27

* CC API Endpoint Version  
2.58.0

* CF Trace, if issue relates to HTTP requests or strange behavior  
See https://gist.github.com/twoseat/7dab66b7aa85074feacec76d636cf9c3

* Platform & Shell Details  
macOS Sierra and stock Terminal

* Any other relevant information
Running against pcfdev 0.21

**Comment by dkoper** [comment]
CC: @shalako 

@twoseat As the purpose of a route service is to add preprocessing to application requests, if a route is only bound to a route service and not to an app, it is basically orphaned (in a way, the whole route service is). Hence such routes should be deleted as well?

**Comment by shalako** [comment]
Seems like either:

- routes bound to services but not apps are considered orphaned, and the command should delete the route and leave the service instance, requiring the cli to unbind the service first. the user should not receive a request to unbind them. 

or 

- routes bound to services but not apps are not considered orphaned, so the cli should not attempt to delete them, and the user wouldn't see the must-unbind-first error. 

Given current ux I tend to agree with Dies (option 1), but looking forward we may eventually fulfill some interesting use cases in which routes don't have to be mapped to apps. For this reason, and because it's lower impact, while improving current UX, I'm inclined to suggest #2.

**Comment by twoseat** [comment]
I'm a relatively naive user, so my expectation that the service route would be deleted is of limited value. Either of @shalako's options sound reasonable to me, so long as we don't get the current 400!

**Comment by XenoPhex** [comment]
Hi @twoseat thanks for creating this issue. We believe the issue you described above will be fix when we address #746. We will leave this issue open to ensure that will be indeed the case.

**Comment by willmurphyscode** [comment]
This issue was resolved by https://github.com/cloudfoundry/cli/pull/1543 and will be fixed in the next release of the CF CLI.
